src/studio/src/designer/backend/Services/Implementation/TextsService.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.Studio.Designer.Services.Implementation
+{
+    /// <summary>
+    /// Interface for dealing with texts in new format in an app repository.
+    /// </summary>
+    public class TextsService : ITextsService
+    {
+        private readonly IAltinnGitRepositoryFactory _altinnGitRepositoryFactory;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="altinnGitRepositoryFactory">IAltinnGitRepository</param>
+        public TextsService(IAltinnGitRepositoryFactory altinnGitRepositoryFactory)
+        {
+            _altinnGitRepositoryFactory = altinnGitRepositoryFactory;
+        }
+
+        /// <inheritdoc />
+        public async Task<string> GetTextContent(string org, string repo, string developer, string languageCode)
+        {
+            var altinnGitRepository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(org, repo, developer);
```

Comment:
Variable named altinnGitRepository but method returns  altinnAppGitRepository. Since there is a AltinnGitRepository this might lead to confusion when reading code. On the other side - when we get to getting texts in a non-app repository this method needs to check the repository type.